From ee4049d420bdef2f8c2aa08b7c18ac188a5df08a Mon Sep 17 00:00:00 2001
From: Marta Kuczora <kuczoram@cloudera.com>
Date: Mon, 8 May 2017 14:33:19 +0200
Subject: [PATCH 1098/1164] CDH-53519: Use the HdfsAdmin API to get the
 KeyProvider and drop the call of the
 DFSClient.isHDFSEncryptionEnabled method

Change-Id: Ide88812cb1ddfebb350c8f527c12c1b57e7143ee
---
 .../apache/hadoop/hive/shims/Hadoop23Shims.java    |   14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
index 9d1fd32..7e6baad 100644
--- a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
+++ b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
@@ -1261,20 +1261,8 @@ public HdfsEncryptionShim(URI uri, Configuration conf) throws IOException {
       DistributedFileSystem dfs = (DistributedFileSystem)FileSystem.get(uri, conf);
 
       this.conf = conf;
-      this.keyProvider = isEncryptionEnabled(dfs.getClient(), dfs.getConf()) ?
-          dfs.getClient().getKeyProvider() : null;
       this.hdfsAdmin = new HdfsAdmin(uri, conf);
-    }
-
-    private boolean isEncryptionEnabled(DFSClient client, Configuration conf) {
-      try {
-        DFSClient.class.getMethod("isHDFSEncryptionEnabled");
-      } catch (NoSuchMethodException e) {
-        // the method is available since Hadoop-2.7.1
-        // if we run with an older Hadoop, check this ourselves
-        return !conf.getTrimmed(DFSConfigKeys.DFS_ENCRYPTION_KEY_PROVIDER_URI, "").isEmpty();
-      }
-      return client.isHDFSEncryptionEnabled();
+      this.keyProvider = hdfsAdmin.getKeyProvider();
     }
 
     @Override
-- 
1.7.9.5

